### PR TITLE
ref(discover) Simplify decodeScalar() and decodeList()

### PR DESCRIPTION
**Pull request by markstory**
Instead of returning string|undefined we can return '' and have a fallback parameter that lets us remove a bunch of `??` usage.

### CODE REVIEW

**Review by k-fish — COMMENTED** [review]
Nice 👏 . Looks like tsc error though.

**Comment by k-fish** [comment]
Look like it might be a real error?

File: src/sentry/static/sentry/app/utils/queryString.tsx [hunk 1]
```diff
@@ -51,26 +51,26 @@ export function appendTagCondition(
   return currentQuery;
 }
 
-export function decodeScalar(
-  value: string[] | string | undefined | null
-): string | undefined {
+// This function has multiple signatures to help with typing in callers.
+export function decodeScalar(value: QueryValue): string | undefined;
```
Comment (k-fish) [inline src/sentry/static/sentry/app/utils/queryString.tsx:55]: Yeah these overloads make the types nicer.

**Review by k-fish — APPROVED** [review]

File: src/sentry/static/sentry/app/utils/queryString.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/utils/queryString.tsx:55]: Using an overload only came to mind after struggling with the tests for a few minutes, and then seeing the fixes causing other failures. :blush:

**Pull request merged**